src/keboola_mcp_server/tools/workspace.py

```
@@ -420,3 +420,24 @@ async def get_quoted_name(self, name: str) -> str:
     async def get_sql_dialect(self) -> str:
         workspace = await self._get_workspace()
         return workspace.get_sql_dialect()
+
+
+class ProjectManager:
+    STATE_KEY = 'project_manager'
+
+    @classmethod
+    def from_state(cls, state: Mapping[str, Any]) -> 'ProjectManager':
+        instance = state[cls.STATE_KEY]
+        assert isinstance(instance, ProjectManager), f'Expected ProjectManager, got: {instance}'
+        return instance
+
+    def __init__(self, client: KeboolaClient):
+        self.project = None
+        self.client = client
+
+    async def get_project_id(self) -> str | int:
+        if self.project is None:
+            # This should probably be handled for exceptions somehow - unclear how at this time
```

I'm not a big fan of these todo comments. I think if we cannot retrieve the project ID, we should throw an exception, because it should not happen and without the project ID, you cannot construct a working link. Alternatively, you can catch exceptions and log an error here and return `None`. But then you'd need to handle it in the tools and throw an exception there.

---

Project manager is gone altogether and tools will always fetch the project id.